### DISCUSSION TITLE
How to use a global endpoint_url for all AWS CLI profiles for path addressing_style ?

### DISCUSSION THREAD

**Discussion by sebma** [aws/aws-cli#8991] [Q&A]
Hi, we are using the path [`addressing_style`](https://docs.aws.amazon.com/cli/latest/topic/s3-config.html#addressing-style) and I'd like to know how to specify the same endpoint_url for all AWS CLI profiles ?
-----
SebM.

**Comment by uldyssian-sh** [comment]
You cannot define a single “global” endpoint_url in ~/.aws/config that automatically applies to all profiles without referencing it.

AWS CLI configuration works per profile. The [default] profile does NOT inherit into other profiles.

You have two correct and supported options.

Option 1 (recommended if you truly want it global): environment variables

Set the endpoint once and it applies to all profiles in that shell/session:

export AWS_ENDPOINT_URL_S3=https://my-s3-compatible.example.com

You can also set the addressing style globally:

export AWS_S3_ADDRESSING_STYLE=path

This is the only way to apply an endpoint to all profiles without editing each one.

Option 2 (config-based, explicit but clean)

Define a shared service and reference it from each profile:

[profile profile1]
services = my-s3
s3 =
  addressing_style = path

[profile profile2]
services = my-s3
s3 =
  addressing_style = path

[services my-s3]
s3 =
  endpoint_url = https://my-s3-compatible.example.com

This keeps the endpoint defined in one place, but each profile must opt in.

Summary:

AWS CLI has no true “global profile inheritance”.
Use environment variables for a truly global endpoint.
Use a shared [services] block if you want a config-file solution.

**Reply by uldyssian-sh** [reply]
Thanks! If this helps, please consider marking it as “Helpful” or “Accepted” 😊
